go/enclave/rollupchain/rollup_chain.go

```diff
@@ -697,3 +675,20 @@ func (rc *RollupChain) GetBalance(encryptedParams common.EncryptedParamsGetBalan
 
 	return encryptedBalance, nil
 }
+
+func (rc *RollupChain) signRollup(r *obscurocore.Rollup) {
+	var err error
+	h := r.Hash()
+	r.Header.R, r.Header.S, err = ecdsa.Sign(rand.Reader, rc.privateKey, h[:])
+	if err != nil {
+		log.Panic("Could not sign rollup. Cause: %s", err)
+	}
+}
+
+func (rc *RollupChain) verifySig(r *obscurocore.Rollup) bool {
+	h := r.Hash()
+	if r.Header.R == nil || r.Header.S == nil {
+		panic("Missing signature on rollup")
+	}
+	return ecdsa.Verify(&rc.privateKey.PublicKey, h[:], r.Header.R, r.Header.S)
```

Comment:
I think I've misunderstood something here. If you're signing the headers with the enclave's own private keys then won't some of these rollups have been signed by different enclaves and you'll need to lookup their public keys from somewhere (stored them from their attestations).

Or should these headers be signed with the network's private key?

Reply:
good point. I'll raise a ticket and add a todo

Reply:
I raised: https://github.com/obscuronet/obscuro-internal/issues/587